popup.js

```diff
@@ -1,32 +1,63 @@
-(function (MemoryGame, $) {
+(function (MemoryGame, EventDispatcher, $) {
 
   /**
    * A dialog for reading the description of a card.
+   * @see https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/
    *
    * @class H5P.MemoryGame.Popup
+   * @extends H5P.EventDispatcher
    * @param {H5P.jQuery} $container
    * @param {Object.<string, string>} l10n
    */
   MemoryGame.Popup = function ($container, l10n) {
+    // Initialize event inheritance
+    EventDispatcher.call(this);
+
     /** @alias H5P.MemoryGame.Popup# */
     var self = this;
 
     var closed;
 
-    var $popup = $('<div class="h5p-memory-pop" role="dialog"><div class="h5p-memory-top"></div><div class="h5p-memory-desc h5p-programatically-focusable" tabindex="-1"></div><div class="h5p-memory-close" role="button" tabindex="0" title="' + (l10n.closeLabel || 'Close') + '" aria-label="' + (l10n.closeLabel || 'Close') + '"></div></div>').appendTo($container);
-    var $desc = $popup.find('.h5p-memory-desc');
-    var $top = $popup.find('.h5p-memory-top');
+    const $popup = $(
+      '<div class="h5p-memory-obscure-content"><div class="h5p-memory-pop" role="dialog" aria-modal="true"><div class="h5p-memory-top"></div><div class="h5p-memory-desc h5p-programatically-focusable" tabindex="-1"></div><div class="h5p-memory-close" role="button" tabindex="0" title="' + (l10n.closeLabel || 'Close') + '" aria-label="' + (l10n.closeLabel || 'Close') + '"></div></div></div>'
+      )
+      .on('keydown', function (event) {
+        if (event.code === 'Escape') {
+          self.close(true);
+          event.preventDefault();
+        }
+      })
+      .hide()
+      .appendTo($container);
+
+    const $top = $popup.find('.h5p-memory-top');
 
     // Hook up the close button
-    $popup.find('.h5p-memory-close').on('click', function () {
-      self.close(true);
-    }).on('keypress', function (event) {
-      if (event.which === 13 || event.which === 32) {
+    const $closeButton = $popup
+      .find('.h5p-memory-close')
+      .on('click', function () {
         self.close(true);
-        event.preventDefault();
-      }
+      })
+      .on('keydown', function (event) {
+        if (event.code === 'Enter' || event.code === 'Space') {
+          self.close(true);
+          event.preventDefault();
+        }
+        else if (event.code === 'Tab') {
+          event.preventDefault(); // Lock focus
+        }
     });
 
+    const $desc = $popup
+      .find('.h5p-memory-desc')
+      .on('keydown', function (event) {
```

Comment:
I do not see the reason of this, because it seems we set `tabindex="-1"` for description so this will be never focused. Or please explain where this code will execute?

Reply:
@ravimajithia That description will receive focus (that's why it needs `tabindex="-1"` in the first place) when the overlay is opened and the focus trap sets the focus inside the dialog. See https://github.com/h5p/h5p-memory-game/pull/80/files/4fab5bb8bbf213efd3361f8173c997f5e2657e7f#diff-3df6958c77d615b266b54fcf12fadf313e7c1e4b168f0c61ea8c3424eb39eb8cR5 for why that is done. If the `Tab` key was not checked here, focus would not remain inside the dialog.

Reply:
Just wanted to make sure that the link you are referring is point to this line -
`* @see https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/` is this correct

Reply:
@ravimajithia Yup. https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/ holds information about how a modal dialog should be structured. In particular, the "Note" describes that it's possible to set focus to a non-actionable element with `tabindex="-1"` if there's no other focussable element inside the modal - that's what is the case here.

Reply:
Great! Thanks for explaining!